bite-size-notes/Week6C-OptimalControl-MohammadT.md

```diff
@@ -0,0 +1,148 @@
+
+
+#### Example:
+
+Consider a robotic entity traversing through a grid-based environment, facing decisions at each grid cell regarding movement direction. The Bellman equation empowers the robot to discern the optimal action to undertake at each cell, weighing the immediate cost of movement against the anticipated cost of reaching the goal from the subsequent cell. By iteratively unraveling the Bellman equation, the robot can chart the optimal path to its destination, while minimizing the overall cost incurred.
+
+### Linear Quadratic Regulator (LQR) Control
+
+LQR control emerges as a potent technique for stabilizing linear systems characterized by quadratic cost functions. Its primary objective revolves around minimizing a cost function subject to state space of the system where x signifies the system state, u denotes the control input, Q represents a positive definite weighting matrix for the state, and R stands as a positive definite weighting matrix for the control input.
+
+The optimal control law for LQR control is delineated by u* = -kx wherein K denotes the solution to the associated algebraic Riccati equation:
+
+$$ \ A^T P + PA - PB(R + B^T P B)^{-1} B^T P + Q = 0 \ $$
+
+Once K is ascertained, it can be wielded to compute the optimal control input u* for any bestowed state x, thereby ensuring stability and optimal performance.
+
+#### Example:
+
+Consider an inverted pendulum system tasked with maintaining the pendulum in an upright position. LQR control calculates the optimal control input (typically a force) requisite to stabilize the pendulum around the upright orientation. By modulating the control input predicated on the prevailing state of the pendulum, LQR control adeptly upholds stability even amidst perturbations.
+
+### HJ Reachability
+
+HJ reachability represents a mathematical framework instrumental in analyzing safe regions within dynamical systems constrained by limits. This framework entails solving the Hamilton-Jacobi partial differential equation (PDE):
+
+$$ \ \frac{\partial V}{\partial t} + H(x, \nabla V) = 0 \ $$
+
+Here, V embodies the value function denoting the optimal cost-to-go from a conferred state, t designates time, x signifies the state vector, and nabla V encapsulates the gradient of V with respect to x. The function 
+$$ \(H(x, \nabla V)\) $$
+dubbed the Hamiltonian, encapsulates the system's dynamics.
+
+By solving the HJ PDE, one can elicit the value function V(x), furnishing insights into the reachable set of the system. This set encompasses all states accessible from a given initial state while satisfying specific safety constraints.
+
+The reachable set materializes from the level sets of the value function V, demarcating regions in the state space wherein the system can safely operate without contravening constraints.
```

Comment:
My favorite physics teacher would say that it's not necessarily a bad thing in STEM to use more standard descriptor verbs when explaining concepts.

Even though these verbs are more complex and thus _can be_ more descriptive, it's starting to disrupt the educational value of the notes with how many uncommon synonyms are being utilized. Wouldn't want to have a reader stop to look up a less used verb.

Example:
"Contravening" is an excellent verb to use English-wise, but the most common synonym used with constraints would be "violating".

Not going to recommend any changes unless you are in agreement: I understand it could be a writing style. But I feel it would be remiss not to mention this, it's excellent practice for paragraphs where you are explaining impacts and need flair, but it _could_ backfire on you elsewhere.

Reply:
I appreciate your feedback and totally agree with the point about using more standard descriptor verbs, especially in educational or technical contexts. It's essential to prioritize clarity and accessibility for readers, so using commonly understood terms like "violating" instead of less common synonyms is definitely a good practice. I tried to modify it.